python/hail/context.py

```diff
@@ -714,6 +714,27 @@ def import_vcf(self, path, force=False, force_bgz=False, header_file=None, min_p
         return VariantDataset(self, jvds)
 
     @handle_py4j
+    @record_method
+    @typecheck_method(path=oneof(strlike, listof(strlike)),
+                      min_partitions=nullable(integral),
+                      drop_samples=bool)
+    def import_matrix(self, path, min_partitions = None, drop_samples = False):
```

Comment:
This is a good start.

I think we should also support at least 5 options for the cell type: Int{32, 64}, Float{32, 64} and String (so the users can use `annotate_genotypes` to parse the cells themselves.)  Second, you should support missingness (see `HailContext.import_table`'s `missing` option.